src/test/java/org/ject/support/external/s3/S3ServiceTest.java

```diff
@@ -29,31 +30,63 @@ class S3ServiceTest {
     @InjectMocks
     private S3Service s3Service;
 
-    @Test
-    @DisplayName("create pre-signed url")
-    void create_presigned_url() throws MalformedURLException {
-        // given
-        Long memberId = 123L;
-        String fileName = "test.pdf";
-        String expectedKeyName = String.format("%s/%s", memberId, "uuid_test.pdf");
-        String expectedUrl = String.format("%s%s", "https://s3.test.com/", expectedKeyName);
-        Instant expirationTime = Instant.now().plusSeconds(600);
+    private TestParameter testParameter;
 
+    @BeforeEach
+    void setUp() throws MalformedURLException {
+        testParameter = new TestParameter(123L, "test.pdf", Instant.now().plusSeconds(600));
         PresignedPutObjectRequest mockRequest = mock(PresignedPutObjectRequest.class);
-        when(mockRequest.url()).thenReturn(URI.create(expectedUrl).toURL());
-        when(mockRequest.expiration()).thenReturn(expirationTime);
-
+        when(mockRequest.url()).thenReturn(URI.create(testParameter.expectedUrl).toURL());
+        when(mockRequest.expiration()).thenReturn(testParameter.expirationTime);
```

Comment:
어차피 단순 데이터 조회의 기능밖에 하지 않기 때문에 요청 객체의 getter를 mocking하기 보다는 mock 객체를 실제로 만드는것이 바람직하다고 생각합니다

Reply:
조금만 더 자세히 말씀해주실 수 있나요?? 이해가 안 가요🫠

Reply:
```suggestion
        //when(mockRequest.url()).thenReturn(URI.create(testParameter.expectedUrl).toURL());
        //when(mockRequest.expiration()).thenReturn(testParameter.expirationTime);
        PresignedPutObjectRequest mockRequest = new PresignedPutObjectRequest("test.pdf",time);
```
이런식으로 객체를 실제 생성자 호출로 만들어 테스트함이 불필요한  mocking을 피하는 방법이라고 생각합니다

Reply:
아 이해했어요. 근데 S3Presigner를 통해서만 생성할 수 있는걸로 보입니다!